dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/LoopUnrolling.java

```diff
@@ -137,6 +141,13 @@ private void unrollLoop(CondJump loopBackJump, int bound) {
                 endOfLoopMarker.addFilters(Tag.NOOPT);
                 boundEvent.getPredecessor().insertAfter(endOfLoopMarker);
 
+                // Unrolling is currently run after compilation, thus we need to assign both oId and cId.
+                // The final step of run takes care of assigning uId.
+                for(Event e : Arrays.asList(threadExit, boundEvent, endOfLoopMarker)) {
+                    e.setOId(loopBackJump.getOId());
+                    e.setCId(loopBackJump.getCId());    
```

Comment:
It doesn't make sense to copy metadata to `threadExit` cause that event does not get created here: it already exists.